packages/core/permissions/lib/engine/hooks.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const { cloneDeep, has } = require('lodash/fp');
+const { hooks } = require('@strapi/utils');
+
+const domain = require('../domain');
+
+/**
+ * Create a hook map used by the permission Engine
+ */
+const createEngineHooks = () => ({
+  'before-format::validate.permission': hooks.createAsyncBailHook(),
```

Comment:
I think we should either use before/after or pre/post rather than mixing the two. I'm fine with either set though.

Reply:
Nice catch, thanks!